docs/docs/reference/metaprogramming/macros.md

```diff
@@ -414,33 +414,26 @@ assume that both definitions are local.
 
 The `inline` modifier is used to declare a `val` that is
 either a constant or is a parameter that will be a constant when instantiated. This
-aspect is also important for macro expansion.  To illustrate this,
-consider an implementation of the `power` function that makes use of a
-statically known exponent:
+aspect is also important for macro expansion.
+
+To get values out of expressions containing constants `Expr` provides the methods
+`value` (or `getValue`). This will convert the `Expr[T]` into a `T` (or `Some[T]`) when the
+expression contains value. Otherwise it will emit an error (or retrun `None`).
+To avoid having incidental val bindings generated by the inlining of the `def`
+it is recomended to use an inline parameter. To illustrate this, consider an
+implementation of the `power` function that makes use of a statically known exponent:
 ```scala
-inline def power(inline n: Int, x: Double) = ${ powerCode(n, 'x) }
+inline def power(x: Double, inline n: Int) = ${ powerCode('x, 'n) }
+
+private def powerCode(x: Expr[Double], n: Expr[Int]): Expr[Double] =
+  powerCode(x, n.value) // Transform `n` into an Int or emit an error if it is not possible
```

Comment:
Maybe do it safely by `n.getValue`

Reply:
This has the same semantics as before. If the inline parameter is not a constant, an error is emitted.